### PR TITLE
Dev cli sessions

### PR DESCRIPTION
**Pull request by i-davydenko**
[hld:](https://github.com/sonic-net/SONiC/pull/1367)
[hostcfgd submodule](https://github.com/i-davydenko/sonic-host-services/pull/1)

#### Why I did it
Give ability to:
- configure limit for active login sessions.
- configure ssh-server / serial console autologout timeout 
- configure sysrq-capabilities (enable / disable)

##### Work item tracking
- Microsoft ADO **(number only)**:

#### How I did it
Add new service that responsible for serial configuration;
Update existing flows for extended ssh-server configurations in hostcfgd;
Add YANG model to support new configuration.

#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [ ] 202106
- [ ] 202111
- [ ] 202205
- [ ] 202211
- [ ] 202305

#### Tested branch (Please provide the tested image version)

- [ ] 
- [ ] 

#### Link to config_db schema for YANG module changes

[ssh_server](https://github.com/sonic-net/sonic-buildimage/blob/master/src/sonic-yang-models/doc/Configuration.md#ssh_server)
[serial_console](https://github.com/sonic-net/sonic-buildimage/blob/master/src/sonic-yang-models/doc/Configuration.md#serial_console)

### CODE REVIEW

File: files/build_templates/init_cfg.json.j2 [hunk 1]
```diff
@@ -135,6 +135,18 @@
             "special_class": "true"
         }
     },
+    "SSH_SERVER": {
```
Comment (dgsudharsan) [inline files/build_templates/init_cfg.json.j2]: Please don't define defaults here. Previously it was discussed and removed. Refer the link below. Please define your defaults in the 

``` https://github.com/sonic-net/sonic-buildimage/pull/14513#discussion_r1246974269 ```

File: src/sonic-host-services
Comment (dgsudharsan) [inline src/sonic-host-services:1]: Please don't include submodule updates when raising to sonic-net unless there is some cyclic dependency

File: files/image_config/cli_sessions/serial-config.sh [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+SYSRQ_CONF=0
+# update sysrq for current boot.
+sysrq_conf=`sonic-db-cli CONFIG_DB HGET "SERIAL_CONSOLE|POLICIES" sysrq_capabilities`
+if [ ${sysrq_conf} = "enabled" ]; then
+        SYSRQ_CONF=1
+fi
+sudo echo $SYSRQ_CONF > /proc/sys/kernel/sysrq
+
+# generate env file for profile.d to set auto-logout timeout for serial consoles.
+sonic-cfggen -d -t /usr/share/sonic/templates/tmout-env.sh.j2 > /etc/profile.d/tmout-env.sh
```
Comment (dgsudharsan) [inline files/image_config/cli_sessions/serial-config.sh:15]: Can you clarify how and when this tmout-env.sh is executed?

File: src/sonic-host-services-data/templates/sshd_config [hunk 3]
```diff
@@ -0,0 +1,121 @@
+#	$OpenBSD: sshd_config,v 1.101 2017/03/14 07:19:07 djm Exp $
```
Comment (dgsudharsan) [inline src/sonic-host-services-data/templates/sshd_config]: Where is this file used? Is this added by mistake?

File: src/sonic-yang-models/yang-models/sonic-serial-console.yang [hunk 4]
```diff
@@ -0,0 +1,37 @@
+//filename:  sonic-serial-console.yang
```
Comment (dgsudharsan) [inline src/sonic-yang-models/yang-models/sonic-serial-console.yang:1]: Please add unit test for this file. Refer to other yang model PRs for example

File: src/sonic-yang-models/yang-models/sonic-ssh-server.yang [hunk 5]
```diff
@@ -40,6 +45,20 @@ module sonic-ssh-server {
                         }
                     }
                 }
+                leaf inactivity_timeout {
```
Comment (dgsudharsan) [inline src/sonic-yang-models/yang-models/sonic-ssh-server.yang:48]: Please add UT for these changes

**Review by dgsudharsan — CHANGES_REQUESTED** [review]

File: files/image_config/cli_sessions/serial-config.sh (hunk 2 above)
Comment (i-davydenko) [reply files/image_config/cli_sessions/serial-config.sh:15]: _tmout-env.sh_ (and all other placed under /etc/profile.d/) runned on each new shell session startup

File: src/sonic-host-services-data/templates/sshd_config (hunk 3 above)
Comment (i-davydenko) [reply src/sonic-host-services-data/templates/sshd_config]: yes, this was a mistake. removed.

**Review by dgsudharsan — APPROVED** [review]

**Pull request merged**